editor/plugins/texture_region_editor_plugin.cpp

```diff
@@ -621,19 +621,28 @@ void TextureRegionEditor::apply_rect(const Rect2 &p_rect) {
 }
 
 void TextureRegionEditor::_update_rect() {
-	if (node_sprite)
-		rect = node_sprite->get_region_rect();
-	else if (node_sprite_3d)
-		rect = node_sprite_3d->get_region_rect();
-	else if (node_ninepatch) {
+	if (node_sprite) {
+		if (node_sprite->is_region()) {
+			rect = node_sprite->get_region_rect();
+		} else {
+			rect = Rect2(Vector2(), node_sprite->get_texture()->get_size());
+		}
+	} else if (node_sprite_3d) {
+		if (node_sprite_3d->is_region()) {
+			rect = node_sprite_3d->get_region_rect();
+		} else {
+			rect = Rect2(Vector2(), node_sprite_3d->get_texture()->get_size());
+		}
+	} else if (node_ninepatch) {
```

Comment:
Hey @bojidar-bg correct me if I'm wrong here, but these are kinda ad hoc statements, no?

Is it possible, in C++, to check if there is a `rect_region` property in the currently selected node? This way this would cover all possible cases where a `rect_region` should be defined. Then inside that we could check if there is a `region_eneabled` property and if this property is true.

I'm pretty sure we don't have this in C++, but it worth asking :sweat_smile:

Reply:
These statements were like that before I changed the code, so no need to fix them (yet).
It is possible to use `Object::has_property` and `Object::get` here, indeed, but it is a bit less performant, and has the side effect of working with arbitrary unrelated objects which just happen to have a `region_rect` property defined (perhaps through script).

Reply:
Interesting, thanks for the enlighting answer :smile: